contentcuration/contentcuration/views.py

```diff
@@ -270,6 +273,54 @@ def _duplicate_node(node, sort_order=1, parent=None):
 
     return new_node
 
+
+def move_nodes(request):
+    logging.debug("Entering the move_nodes endpoint")
+
+    if request.method != 'POST':
+        raise HttpResponseBadRequest("Only POST requests are allowed on this endpoint.")
+    else:
+        data = json.loads(request.body)
+
+        try:
+            nodes = data["nodes"]
+            target_parent = ContentNode.objects.get(pk=data["target_parent"])
+            channel_id = data["channel_id"]
+        except KeyError:
+            raise ObjectDoesNotExist("Missing attribute from data: {}".format(data))
+
+        with transaction.atomic():
+            for n in nodes:
+                node = ContentNode.objects.get(pk=n['id'])
+                _move_node(node, parent=target_parent, sort_order=n['sort_order'], channel_id=channel_id)
+
+        return HttpResponse(json.dumps({
+            "success": True,
+            "nodes": [n['id'] for n in nodes]
+        }))
+
+def _move_node(node, parent=None, sort_order=1, channel_id=None):
+    node.parent = parent
+    node.sort_order = sort_order
+    node.changed = True
+    descendants = node.get_descendants(include_self=True)
+    node.save()
+
+    for tag in ContentTag.objects.filter(tagged_content__in=descendants):
```

Comment:
May want to add `.distinct()` after `filter(...)`, otherwise tags that are associated with multiple descendants may get processed twice (shouldn't cause problems, just inefficient).